PhysicsTools/ONNXRuntime/src/ONNXRuntime.cc

```diff
@@ -104,6 +132,10 @@ namespace cms::Ort {
       } else {
         input_dims = input_shapes[input_pos];
         // rely on the given input_shapes to set the batch size
+        if (input_dims[0] != batch_size) {
+          throw cms::Exception("RuntimeError") << "The first element of `input_shapes` (" << input_dims[0]
+                                               << ") does not match the given `batch_size` (" << batch_size << ")";
+        }
```

Comment:
this change is unrelated to the use of GPUs, isn't it ?

Reply:
No it's unrelated to GPU -- just a helpful error message for users.